www/perl-lib/SMT/RegistrationSharing.pm

```diff
@@ -243,42 +218,41 @@ sub shareRegistration
             . '</tableData>';
     }
     $regXML .= '</registrationData>';
+    $dbh->disconnect();
+    _sendRegData($regXML, $log);
+}
 
-    # Send the registration data to the configured SMT siblings
-    if ( $regXML =~ /GUID/) {
-        my @smtSiblings = split /,/, $shareRegDataTargets;
-        for my $smtServer (@smtSiblings) {
-            my $url = "https://$smtServer/center/regsvc"
-                . '?command=shareregistration'
-                . '&lang=en-US&version=1.0';
-            my $response = _sendData($url, $regXML, $log);
-            if (! $response ) {
-                return;
-            }
-            if (! $response->is_success ) {
-                my $msg = $response->message;
-                my $details = $response->content;
-                my $guidMsg = "Could not share registration for $regGUID";
-                my $responseMsg = "Response: $msg";
-                my $detailsMsg = "Response: $details";
-                if ($log) {
-                    print $log $guidMsg . "\n";
-                    print $log $responseMsg . "\n";
-                    print $log $detailsMsg . "\n";
-                } else {
-                    $apache->warn($guidMsg);
-                    $apache->warn($responseMsg);
-                    $apache->warn($detailsMsg);
-                    _logShareRecord($smtServer,$url,$regXML);
-                }
-
-            } else {
-                _sharePreviousRegistrations($smtServer, $log);
-            }
-        }
+#
+# Share a single product registration. In the SCC paradigm products/modules
+# are registered one at a time and we do not know when the client considers
+# registration to be complete. Thus we share every product as a registration
+#
+sub shareProductRegistration
+{
+    my $regGUID = shift;
+    my $prodID = shift;
+    my $apache = Apache2::ServerUtil->server;
+    my $dbh = SMT::Utils::db_connect();
+    my $regXML = '<?xml version="1.0" encoding="UTF-8"?>'
+        . '<registrationData>';
+    # Get the registration data
+    my $statement = sprintf("SELECT * from Registration where GUID=%s and PRODUCTID=%s",
+                            $dbh->quote($regGUID),
+                            $dbh->quote($prodID));
```

Comment:
OK, one of the requirements of the infrastructure is that all SMT servers are setup identically. However, now this explains why smt-list-registrations on one of my test setups did not work as expected, i.e. the product did not show up in the listing, although failover still worked on the client :) .

I will modify the code to take the data from the Products table into account. Such that the PRODUCTID is the appropriate value in the Registration table for that server rather than being the value from the sibling.

If I found the correct connection than the ID column of the Products table matches the PRODUCTID column of the Registration table.